akka-stream/src/main/scala/akka/stream/scaladsl/Flow.scala

```diff
@@ -1340,6 +1341,38 @@ trait FlowOps[+Out, +Mat] {
         FlowShape(bcast.in, bcast.out(0))
     })(matF)
 
+  // TODO find this buddy a home (sounds useful enough to expose?)
+
+  def exposeAttribute[A <: Attribute]()(implicit t: ClassTag[A]): Repr[(Out, Option[A]), Mat] =
+    exposeAttribute(ConstantFun.scalaTuplify[Out, Option[A]])(t)
+
+  def exposeAttribute[A <: Attribute, E](exposeAs: (Out, Option[A]) ⇒ E)(implicit t: ClassTag[A]): Repr[E, Mat] =
+    this.via(Flow.fromGraph(new ExposeAttributeStage[Out, A, E, Mat](t.runtimeClass.asInstanceOf[Class[A]], exposeAs)))
+
+  class ExposeAttributeStage[In, A <: Attribute, E, M](attrClazz: Class[A], exposeFn: (In, Option[A]) ⇒ E)
```

Comment:
The problem is "have http pipeline reusable" this address is injected though, so we need a way to do that, and pull it out once the stream runs. Attributes are a way of doing that; another one would be inventing what "parameterized materialization" would mean, and where the value could be pulled out from then.

Specifically it's about:

>  The remote address of the peer is currently passed to the parser during creation, removing this obstacle would need a method to parameterize materialization. The suggested workaround of materializing a function or promise (or any kind of cell) to pass the value in after materialization is again prone to race conditions (or would complicate the target stage with extra waiting code to receive the value).

from: https://github.com/akka/akka/issues/19050

Other ideas?